lib/p2p/Peer.ts

```diff
@@ -121,8 +123,8 @@ class Peer extends EventEmitter {
   /**
    * Creates a Peer from an inbound socket connection.
    */
-  public static fromInbound = (socket: Socket, logger: Logger): Peer => {
-    const peer = new Peer(logger, addressUtils.fromSocket(socket));
+  public static fromInbound = (socket: Socket, logger: Logger, discover: boolean, discoverminutes: number): Peer => {
```

Comment:
perhaps pass along the entire Pool config object. will be easier to follow when it's being read from (`this.discover` vs `this.config.discover`), and easier to add more fields later. 
It might require to move the `PoolConfig` type to a different location. If it's getting complicated you can leave it as is for now.

Reply:
passing config into it from `Pool` I'm not the one to decide where to put config type, therefore leaving it as is, but changing signature of Peer constructor to accept PoolConfig